code/Cols.py

```diff
@@ -0,0 +1,31 @@
+from Num import Num
+from Sym import Sym
+from Utils import Utils
+
+
+class Cols:
+
+    def __init__(self, names):
+        self.x = {}  # independent columns (that are not skipped)
+        self.y = {}  # depedent columns (that are not skipped)
+        self.all = {}  # all the columns (including the skipped ones)
+        self.klass = None  # the single dependent klass column (if it exists)
+        self.names = names  # all column names
+        self.process_names(names)
+
+    def process_names(self, names):
+        for index, value in enumerate(names):
+            # column that starts with upper case alphabet is num else sym
+            col = Num(index, value) if value[0].isalpha() and value[0].isupper() else Sym(index, value)
+            Utils.push(self.all, col)
+            # print(self.all)
+            if not value.endswith(':'):  # columns ending with ':' are skipped
+                input_dict = self.x if any(value.endswith(i) for i in ['+', '!', '-']) else self.y
```

Comment:
here string.find is different from string.endswith. Please replace it with something similar to find functionality. Lua tutorial here: https://www.tutorialspoint.com/string-find-function-in-lua

Reply:
Also, if the string finds +,!,- then it should be assigned to self.y and not self.x, https://github.com/timm/l5/blob/c0a035598d25234d5b8083fda94742d100fe7d3c/src/csv.lua#L144

Reply:
Add this line as well: https://github.com/timm/l5/blob/c0a035598d25234d5b8083fda94742d100fe7d3c/src/csv.lua#L145